src/dnaio/_core.pyx

```diff
@@ -433,11 +401,11 @@ cdef class FastqIter:
             self.buffer = tmp
         else:
             # Move the incomplete record from the end of the buffer to the beginning.
-            remaining_bytes = self.bytes_in_buffer - self.record_start
+            remaining_bytes = self.bytes_in_buffer - (self.record_start - self.buffer)
             # Memmove copies safely when dest and src overlap.
-            memmove(self.buffer, self.buffer + self.record_start, remaining_bytes)
+            memmove(self.buffer, self.record_start, remaining_bytes)
             self.bytes_in_buffer = remaining_bytes
-            self.record_start = 0
+        self.record_start = self.buffer
```

Comment:
Did you de-dent this on purpose?

Reply:
Yes. There are two blocks. The above block resizes buffer. This means that `self.buffer` can point to a different memory adress.

Since self.record_start is now a pointer itself rather than an offset, it needs to be set to the start of the buffer (self.buffer) in both cases.